src/services/httpApi.service.ts

```diff
@@ -0,0 +1,38 @@
+import axios, { AxiosRequestConfig } from 'axios';
+
+enum HttpResult {
+  SUCCESS,
+  FAIL,
+}
+
+interface IHttpResponse {
+  error: string | null;
+  data: any | null;
+  result: HttpResult
+}
+
+const httpErrorMessage = (url: string, errorCode: string, errorMessage: string, customMessage?: string) => `Http request to url ${url} ${customMessage} failed with error code ${errorCode}. Error: ${errorMessage}`;
+
+const httpGeneralErrorMessage = (url: string) => `Http request to url ${url} failed.`;
+
+const get = () => {
+};
+
+const put = async (url: string, data?: any, config?: AxiosRequestConfig): Promise<IHttpResponse> => {
+  try {
+    const response = await axios.put(url, data, config);
+    return { error: null, data: response.data, result: HttpResult.SUCCESS };
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
+      console.log(httpErrorMessage(url, error.code!, error.message, `Data: ${JSON.stringify(data)}`));
+      return { error: error.response!.data, data: null, result: HttpResult.FAIL };
```

Comment:
error object on his own gives indication that request was successful (=== null) or not otherwise
thus making `result: HttpResult.FAIL` excessive

Reply:
I think without the HttpResult it is implicit but not enough because according to the types (error: string | null; data: any | null;) they can both be null and can both be not null which is an errorneous state. Data can be an empty string which is falsy.

Reply:
how both error and data can be null?

Reply:
If there is a bug :D